routes/web.php

```diff
@@ -26,7 +26,8 @@
 Route::feeds();
 
 Route::get('/', [IndexController::class, 'show'])->name('home')->middleware(['migration:complete']);
-Route::get('/gallery', [IndexController::class, 'gallery'])->name('gallery')->middleware(['migration:complete']);
+Route::get('/gallery', [IndexController::class, 'gallery'])->name('gallery')->middleware(['migration:complete'])
+		->name('login');
```

Comment:
This is required because for some reason, Auth randomly decides to redirect to Login if the session cannot be authenticated.

Reply:
I don't like "Heisenberg" behavior and your explanation does not fully satisfy me. "Sometimes for some reason" is too vague for my taste. Do you have a more specific explanation?

If not, how can I reproduce the problem. Then I would try to pin down the problem in the step debugger.

Reply:
Moreover, I assume you noticed, that we already assign the name `'gallery'` to the route. Calling `->name` a second time, appends the string, so the route is named `gallerylogin`. I wonder how this solves any problem at all, as a route with the name `gallerylogin` is nowhere called.

Reply:
> I don't like "Heisenberg" behavior and your explanation does not fully satisfy me. "Sometimes for some reason" is too vague for my taste. Do you have a more specific explanation?
> 
> If not, how can I reproduce the problem. Then I would try to pin down the problem in the step debugger.

Create empty sqlite DB, migrate, go to the gallery page, insert login & password & confirm password, Save.
Do another action / refresh page => leads to the exception route `login` not found.

Reply:
> Moreover, I assume you noticed, that we already assign the name `'gallery'` to the route. Calling `->name` a second time, appends the string, so the route is named `gallerylogin`. I wonder how this solves any problem at all, as a route with the name `gallerylogin` is nowhere called.

I did not see that (the double naming).

Reply:
Fixed the problem in https://github.com/LycheeOrg/Lychee/pull/1403/commits/8c441ff61a4e129f533f50791581d05dfb0052b5.

 1. The true fix is in the lines

    https://github.com/LycheeOrg/Lychee/blob/8c441ff61a4e129f533f50791581d05dfb0052b5/app/Http/Controllers/Administration/SettingsController.php#L40-L50
    
    After a user has changed the password, we must reset the user in the session otherwise the next request will be considered unauthenticated. Missing so has also led to an inconsistency in states between the frontend and the backend. After the initial admin login, the frontend considers the admin authenticated, but the backend did not so. This also caused a lot of other troubles. For example, the frontend offered the user the option to log out, but the backend did not handle that well either, because there was no user to be logged out.
    
 2. Nonetheless, it is good to have a route called `'login'` as Laravel expects such a route to exist. I also fixed the double naming.
    
 3. However, there is still one thing which needs to be fixed. The same bug as above also affects the use case when normal user change their password. However, we must only call `Auth::login` for the user if users change their _own_ password, not if the admin changes a user's password.
    
    @ildyria Can you take care of that? I am too tired to fix that now. It must be done in `SettingsController::updateLogin()`.

Reply:
Just some background info. The failing lines were

https://github.com/laravel/framework/blob/07aae35474ab412911566c7ec060ea4853efc13c/src/Illuminate/Session/Middleware/AuthenticateSession.php#L55-L57

The middleware stores the hashed password in the session and checks if the stored password hash in the session equals the password hash of the user in the database. If it does not, the session is logs out which in turn unconditionally throws an `UnauthenticatedException` to trigger a redirection to the landing page. That is why we need to update the user object in the session after it has been modified.

@ildyria I would really appreciate if you tried to figure that out yourself the next time instead of just stating that "something sometimes somehow" went wrong. This could have had the potential to really spit into our faces if we had merged this PR unchecked.

Reply:
> 3. However, there is still one thing which needs to be fixed. The same bug as above also affects the use case when normal user change their password. However, we must only call `Auth::login` for the user if users change their own password, not if the admin changes a user's password.
>
> @ildyria Can you take care of that? I am too tired to fix that now. It must be done in SettingsController::updateLogin().

Fixed.

The `UpdateLogin` is updating the current user. So there is no difference between admin and non admin user. To change the credentials of another user, Admin are using `Actions/User/Save.php`.

Reply:
Fixed the way how authentication errors are reported to the frontend in https://github.com/LycheeOrg/Lychee/pull/1403/commits/c7a386a298cc1653f3b6952120221bb73d2d73ab.